sktime/forecasting/reconcile.py

```diff
@@ -0,0 +1,355 @@
+        "handles-missing-data": False,  # can estimator handle missing data?
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd.Series",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "X_inner_mtype": [
+            "pd.DataFrame",
+            "pd.Series",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],  # which types do _fit, _predict, assume for X?
+        "requires-fh-in-fit": False,  # is forecasting horizon already required in fit?
+        "X-y-must-have-same-index": False,  # can estimator handle different X/y index?
+        "enforce_index_type": None,  # index type that needs to be enforced in X/y
+        "capability:pred_int": False,  # does forecaster implement proba forecasts?
+        "fit_is_empty": False,
+    }
+
+    METHOD_LIST = ["mint_cov", "mint_shrink", "wls_var"]
+
+    def __init__(self, forecaster, method="mint_cov"):
+
+        self.forecaster = forecaster
+        self.method = method
+
+        super(ReconcilerForecaster, self).__init__()
+
+    def _add_totals(self, y):
+        """Add total levels to y, using Aggregate."""
+        from sktime.transformations.hierarchical.aggregate import Aggregator
+
+        return Aggregator().fit_transform(y)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Time series to which to fit the forecaster.
+        fh : ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, default=None
+            Exogenous variables for the base forecaster
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self._check_method()
+
+        # # check the length of index if not hierarchical just return forecaster
+        if y.index.nlevels < 2:
+            self.forecaster_ = self.forecaster.clone()
+            self.forecaster_.fit(y=y, X=X, fh=fh)
+            return self
+
+        # check index for no "__total", if not add totals to y
+        if _check_index_no_total(y):
+            y = self._add_totals(y)
+
+        if X is not None:
+            if _check_index_no_total(X):
+                X = self._add_totals(X)
+
+        # fit forecasters for each level
+        self.forecaster_ = self.forecaster.clone()
+        self.forecaster_.fit(y=y, X=X, fh=fh)
+
+        # now summation matrix
+        self.s_matrix = _get_s_matrix(y)
+
+        # parent child df
+        self.parent_child = _parent_child_df(self.s_matrix)
+
+        # bug in self.forecaster_.predict_residuals() for heir data
+        fh_resid = ForecastingHorizon(
+            y.index.get_level_values(-1).unique(), is_relative=False
+        )
+        self.residuals = y - self.forecaster_.predict(fh=fh_resid, X=X)
+
+        if self.method == "mint_cov":
```

Comment:
instead of having strings, would it make sense to have `shrink` and `diag_only` directly as hyperparameters?
Although the current solution seems better to me if we will have methods here that do not use `_get_g_matrix_mint` or a simple extension, in the future.

Reply:
Yeah I think we should maybe leave it as is then? I agree it will be easier to add methods that use the residuals or training data but not the covariance matrix, e.g. top down historical proportions.

Reply:
ok, makes sense